src/java/neroxis/transformer/MapTransformer.java

```diff
@@ -308,6 +333,24 @@ public void transform() {
             map.getDecals().clear();
             map.getDecals().addAll(getTransformedDecals(decals));
         }
+
+        oldMapSize = map.getSize();
+        if (mapSize == -1) {
+            mapSize = resize;
+        }
+        if (mapSize == -1) {
+            mapSize = oldMapSize;
+        }
+        float sizeRatio = (float) resize / (float) oldMapSize;
+        if (shiftX == 1000000) {
+            shiftX = mapSize / 2;
+        }
+        if (shiftZ == 1000000) {
+            shiftZ = mapSize / 2;
+        }
```

Comment:
set these in argument parsing

Reply:
I would do that if the size was known definitively at that point.

Reply:
made flags